src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/CollectionDeserializationHelpers.cs

```diff
@@ -98,6 +99,13 @@ private static void AddToCollectionCore(this IEnumerable items, IEnumerable coll
                     addMethod.Invoke(collection, new object[] { element });
                 }
             }
+
+            EdmChangedObjectCollection edmchangedObjectCollection = list as EdmChangedObjectCollection;
+
+            if(edmchangedObjectCollection != null)
+            {
+                collection = edmchangedObjectCollection.ChangedObjectCollection;
+            }
```

Comment:
What is this logic for?  It seems to set the collection variable which is never again used in this method.